### PR TITLE
fix file descriptor leaks

### PR DESCRIPTION
**Pull request by rbren**
**End-user friendly description of the problem this fixes or functionality that this introduces**

- [ ] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below
no changelog

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Pretty sure we're leaking FDs thanks to not closing websockets!

---
**Link of any specific issues this addresses**

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:4001192-nikolaik   --name openhands-app-4001192   docker.all-hands.dev/all-hands-ai/openhands:4001192
```

### CODE REVIEW

**Review by xingyaoww — APPROVED** [review]

File: openhands/controller/agent_controller.py [hunk 1]
```diff
@@ -191,7 +192,9 @@ async def start_step_loop(self):
         """The main loop for the agent's step-by-step execution."""
 
         self.log('info', 'Starting step loop...')
-        while should_continue():
+        while True:
+            if self._closed:
+                break
```
Comment (rbren) [inline openhands/controller/agent_controller.py:197]: This is probably the biggest fix 🙈 

These loops were continuing indefinitely even after a session was closed

Comment (xingyaoww) [reply openhands/controller/agent_controller.py:197]: Will the `while True` re-introduce the unable to interrupt issue @tofarr spotted earlier?

Comment (rbren) [reply openhands/controller/agent_controller.py]: woof good catch, that was me debugging. Reverted the `while True` part, kept the `self._closed` part

**Review by xingyaoww — APPROVED** [review]
LGTM!

**Pull request merged**